xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESound.h

```diff
@@ -59,6 +61,7 @@ class CActiveAESound : public IAESound
   static int64_t Seek(void *h, int64_t pos, int whence);
 
 protected:
+  CActiveAE *m_activeAE;
```

Comment:
Actually, I spend some more time thinking about this and the design is completely broken. ActiveAESound should not know about ActiveAE or the other way around. Circular dependencies always suck and cause issues so this needs a redesign to be acceptable.

I would suggest making ActiveAESound and ActiveAEStream observable and have ActiveAE listen to their events. It's possibly an alternative to use the AE messaging system as well to to fix it.

Either way, this needs fixing before merge.